debezium-connector-yugabytedb2/src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java

```diff
@@ -364,12 +374,23 @@ private void getChanges2(ChangeEventSourceContext context,
                                         // too early
                                         if (message.getOperation() == Operation.BEGIN) {
                                             LOGGER.debug("LSN in case of BEGIN is " + lsn);
+
+                                            seenBeginRecord.put(tabletId, Boolean.TRUE);
                                         }
                                         if (message.getOperation() == Operation.COMMIT) {
                                             LOGGER.debug("LSN in case of COMMIT is " + lsn);
                                             offsetContext.updateWalPosition(tabletId, lsn, lastCompletelyProcessedLsn, message.getCommitTime(),
                                                     String.valueOf(message.getTransactionId()), null, null/* taskContext.getSlotXmin(connection) */);
                                             commitMessage(part, offsetContext, lsn);
+
+                                            if (recordsInTransactionalBlock.get(tabletId) == 0) {
+                                                LOGGER.warn("Records in the transactional block for tablet {} are 0", tabletId);
+                                            } else {
+                                                LOGGER.info("Records in the transactional block for tablet {}: {}", tabletId, recordsInTransactionalBlock.get(tabletId));
```

Comment:
At info level there will be too much logs. Most of the time there will be single record transaction.